temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -334,8 +334,17 @@ public <R> StartUpdateOutput<R> startUpdate(StartUpdateInput<R> input) {
             .setRequest(request)
             .build();
     Deadline pollTimeoutDeadline = Deadline.after(POLL_UPDATE_TIMEOUT_S, TimeUnit.SECONDS);
-    UpdateWorkflowExecutionResponse result =
-        genericClient.update(updateRequest, pollTimeoutDeadline);
+
+    // Re-attempt the update until it is at least accepted, or passes the lifecycle stage specified
+    // by the user.
+    UpdateWorkflowExecutionResponse result;
+    do {
+      result = genericClient.update(updateRequest, pollTimeoutDeadline);
```

Comment:
Can you make sure down below in `pollWorkflowUpdateHelper` that you remove the logic that retries on gRPC deadline exceeded error? That should no longer occur, we should just be bubbling all errors out

Reply:
Wait, isn't the Python one doing that when it passes `retry=True` to the service client? Or, if that doesn't retry timeouts, then where is that happening? Because https://github.com/temporalio/sdk-python/blob/1a2acd59634a3b1d694937b8a8433c0014247370/temporalio/client.py#L4303 says it will, but there's no explicit handling of timeouts here: https://github.com/temporalio/sdk-python/blob/1a2acd59634a3b1d694937b8a8433c0014247370/temporalio/client.py#L4359

Reply:
It's easy to change Java to not do this and just default to max timeout for getResult calls, but, not sure that's the right thing to do.

Reply:
Yeah, that's done now. All it's doing is just interpreting the failure code into the right exception type which makes sense to me.